src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +35,15 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToObj(i -> {
+                    int number = numbers.get(i);
+                    return (i % 2 != 0) ? number - 1 : number;
+                })
+                .filter(n -> n % 2 != 0)
+                .mapToInt(Integer::intValue)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException("No such elements in the list"));
```

Comment:
Consider creating a helper called `isNumberEven` and use it.

```suggestion
        return IntStream.range(0, numbers.size())
                .map(index -> !isNumberEven(index) ? numbers.get(index) - 1 : numbers.get(index))
                .filter(i -> !isNumberEven(i))
                .average()
                .orElseThrow(NoSuchElementException::new);
```